core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaStoreReaderThread.java

```diff
@@ -147,9 +141,9 @@ public void doWork() {
           V message = null;
           try {
             message =
-                messageBytes == null ? null : serializer.deserializeValue(messageKey, messageBytes);
+                record.value() == null ? null : serializer.deserializeValue(messageKey, record.value());
           } catch (SerializationException e) {
-            log.error("Failed to deserialize a schema or config update", e);
+            throw new RuntimeException("Failed to deserialize a schema or config update", e);
```

Comment:
You commented on this previously:

> Not part of this patch, but it seems like a bug to ignore this exception and continue below. Wouldn't we then incorrectly see the message as an instance of null and delete the key?

Which I interpreted as "throw an exception instead." Did I misinterpret that?